offload/plugins-nextgen/common/include/ErrorReporting.h

```diff
@@ -216,6 +216,90 @@ class ErrorReporter {
                        getAllocTyName(ATI->Kind).data(), DevicePtr);
 #undef DEALLOCATION_ERROR
   }
+
+  /// Report that a kernel encountered a trap instruction.
+  static void reportTrapInKernel(
+      GenericDeviceTy &Device, KernelTraceInfoRecordTy &KTIR,
+      std::function<bool(__tgt_async_info &)> AsyncInfoWrapperMatcher) {
+    assert(AsyncInfoWrapperMatcher && "A matcher is required");
+
+    uint32_t Idx = 0;
+    for (uint32_t I = 0, E = KTIR.size(); I < E; ++I) {
+      auto KTI = KTIR.getKernelTraceInfo(I);
```

Comment:
Does `llvm::enumerate` work here?

Reply:
Likely, is that better? It has no precedent in Offload rn, maybe worth introducing in one swoop, if it is preferred.

Reply:
Not a huge deal, just tends to be cleaner.

Reply:
I can change it here for the final version, assuming I won't encounter any problems.